.github/workflows/lints.yml

```diff
@@ -17,3 +17,19 @@ jobs:
 
       - name: Run nix-formatter-pack-check
         run: nix build .#checks.x86_64-linux.nix-formatter-pack-check
+
+  lint-py:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install nix
+        uses: cachix/install-nix-action@v25
+
+      - name: Run ruff linter
+        run: nix run 'nixpkgs#ruff' -- check
```

Comment:
in the interest of running the version of ruff we expect to run, shouldn't this point to something in `flake.nix`? Unfortunately I don't think `nix run .#nixpkgs.legacyPackages.x86_64.ruff` works but we could expose ruff in the flake and run that.